crawler/src/main.rs

```diff
@@ -375,7 +375,18 @@ async fn main() -> anyhow::Result<()> {
                         let domain_doc = domains_collection
                             .find_one(doc! { "domain": domain }, None)
                             .await?
-                            .ok_or_else(|| anyhow::anyhow!("missing domain for {url}"))?;
+                            .ok_or_else(|| anyhow::anyhow!("missing domain for {url}"));
+
+                        if domain_doc.is_err() {
+                            let id = match document.get_object_id("_id") {
+                                Ok(id) => id.to_hex(),
+                                Err(_) => "<unknown>".into(),
+                            };
+                            erroneous_docs.insert(id, domain_doc.err().unwrap());
+                            continue;
+                        }
+
+                        let domain_doc = domain_doc.unwrap();
```

Comment:
Here we'd like to avoid using `unwrap()` as much as possible. Since you will need either the value or the error, you can match on it directly:

```rust
let domain_doc = match domain_doc {
    Ok(doc) => doc,
    Err(err) => {
        let id = match document.get_object_id("_id") {
            Ok(id) => id.to_hex(),
            Err(_) => "<unknown>".into(),
        };
        erroneous_docs.insert(id, err);
        continue;
    }
};
```

With the code above, you cannot encounter runtime errors (if the code changes in the future), and the check that `domain_doc.is_err()` is only done once.

Reply:
Hi, thanks so much for explaining the logic behind your changes; it goes a long way toward helping me learn. 🙂 Everything looks good/makes sense to me here!